crates/primitives/receipt/src/lib.rs

```diff
@@ -227,6 +226,12 @@ pub struct FeePayment {
     pub unit: PriceUnit,
 }
 
+impl From<Felt> for FeePayment {
+    fn from(fee: Felt) -> Self {
+        Self { amount: fee, unit: PriceUnit::Wei }
```

Comment:
Why the default to `Wei`? Maybe isolate this into a more explicit function like `felt_to_fee_wei`? Nitpick

Reply:
I think this implementation dates back to before V3. I’ll check if it needs to be updated

Reply:
the PriceUnit value needs to be dynamic and depend on the transaction version. For versions ZERO, ONE, and TWO, it should serialize as `Wei`. For all other versions, it should serialize as `Fri`. This ensures compatibility with differing transaction specifications.

I propose addressing this issue in a separate pull request to keep the changes isolated and easier to review.

Reply:
Sounds good, keeping this conversation open in the meantime as a point of reference

Reply:
you can make a comment in the code / lift this concern to an issue

Reply:
added, see add #385